neo4j-plugins/src/main/java/io/cdap/plugin/transformer/MdmLookupTransformer.java

```diff
@@ -0,0 +1,170 @@
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.InvalidEntry;
+import io.cdap.cdap.etl.api.Transform;
+import io.cdap.cdap.etl.api.TransformContext;
+import io.cdap.plugin.sink.Neo4jDataService;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * MdmLookup plugin to verify if data present in MDM.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("MdmLookup")
+@Description("Verify whether the requested value is present in MDM.")
+public class MdmLookupTransformer extends Transform<StructuredRecord, StructuredRecord> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MdmLookupTransformer.class);
+
+    private final LookupConfig config;
+    private Neo4jDataService dataService;
+    private String lookupColumn;
+    private Driver driver;
+
+    public MdmLookupTransformer(LookupConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) {
+        List<String> lookupData = new ArrayList<>();
+        for (Schema.Field field : input.getSchema().getFields()) {
+            processField(input, lookupData, field, false);
+        }
+
+        final List<String> notFoundList = lookupData.stream()
+            .filter(uuid -> dataService.getUniqueNodeByProperty(lookupColumn, uuid) == null)
+            .collect(Collectors.toList());
+
+        if (!notFoundList.isEmpty()) {
+            notFoundList.forEach(id -> emitter.emitError(
+                new InvalidEntry<>(1, String.format("Not found in MDM. UUID: %s", id), input)));
+        } else {
+            emitter.emit(input);
+        }
+    }
+
+    private void processField(StructuredRecord input, List<String> lookupData,
+                              Schema.Field field, Boolean isChildRecord) {
+
+        if (Schema.Type.ARRAY.equals(field.getSchema().getType())) {
+            for (Schema.Field childField : field.getSchema().getComponentSchema().getFields()) {
+                List<StructuredRecord> structuredRecordList = input.get(field.getName());
+                processField(structuredRecordList.get(0), lookupData, childField, true);
+            }
+        } else if (field.getSchema().getType().isSimpleType() && isChildRecord) {
+            LOG.info("It's simple type");
+            if (field.getName().equals(lookupColumn)) {
+                lookupData.add(processProperty(input, field));
+            }
+        } else {
+            LOG.warn("Unrecognized field type");
```

Comment:
This log is useless as we expect only particular field types to be processed here.

Reply:
removed useless block with log